BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -127,58 +127,23 @@ AppendHashStore (
 {
   EFI_STATUS           Status;
   HASH_STORE_TABLE    *LdrKeyHashBlob;
-  HASH_STORE_TABLE    *OemKeyHashBlob;
-  HASH_STORE_TABLE    *OemKeyHashComp;
-  UINT32               OemKeyHashCompBase;
-  UINT32               OemKeyHashUsedLength;
-  INT32                KeyHashSize;
-  UINT8                AuthInfo[SIGNATURE_AND_KEY_SIZE_MAX];
-  SIGNATURE_HDR       *SignHdr;
-  PUB_KEY_HDR         *PubKeyHdr;
+  UINT8               *OemKeyHashBlob;
+  UINT32               OemKeyHashLen;
   HASH_ALG_TYPE        MbHashType;
 
+  // Request to load at the end of current hash store in memory
+  LdrKeyHashBlob = (HASH_STORE_TABLE *)(UINTN)LdrGlobal->HashStorePtr;
+  OemKeyHashBlob = (UINT8 *)LdrKeyHashBlob + LdrKeyHashBlob->UsedLength;
+  OemKeyHashLen  = LdrKeyHashBlob->TotalLength - LdrKeyHashBlob->UsedLength;

```

Comment:
OemKeyHashLen  calculation is required? It would be populated from Container component.

Reply:
This is the max length allowed.  If LoadComponent returns success, the length field will be updated to the actual length.